internal/docstore/docstore.go

```diff
@@ -314,28 +314,50 @@ func toDriverMods(mods Mods) ([]driver.Mod, error) {
 	// Convert mods from a map to a slice of (fieldPath, value) pairs.
 	// The map is easier for users to write, but the slice is easier
 	// to process.
-	// TODO(jba): check for prefix
 
 	// Sort keys so tests are deterministic.
+	// After sorting, a key might not immediately follow its prefix. Consider the
+	// sorted list of keys "a", "a+b", "a.b". "a" is prefix of "a.b", but since '+'
+	// sorts before '.', it is not adjacent to it. All we can assume is that the
+	// prefix is before the key.
 	var keys []string
 	for k := range mods {
 		keys = append(keys, string(k))
 	}
 	sort.Strings(keys)
 
-	dmods := make([]driver.Mod, len(keys))
-	for i, k := range keys {
+	var dmods []driver.Mod
+	for _, k := range keys {
 		k := FieldPath(k)
 		v := mods[k]
 		fp, err := parseFieldPath(k)
 		if err != nil {
 			return nil, err
 		}
-		dmods[i] = driver.Mod{FieldPath: fp, Value: v}
+		for _, d := range dmods {
```

Comment:
One suggestion, you can do the whole prefix check process in the beginning of this function, I think that way looks more clear as you are just looping through the keys of mods and trying to find all possible prefix and then append them to `dmods` later.

Reply:
It's hard to find the prefixes before I parse the keys into field paths. For example, `a` is a prefix of `ab` and `a.b`, but only the second is wrong.